software_installed.go

```diff
@@ -54,3 +56,28 @@ func (c *ClientIMPL) GetSoftwareInstalled(
 	})
 	return resp, err
 }
+
+func (c *ClientIMPL) GetSoftwareMajorVersion(
+	ctx context.Context) (majorVersion int, err error) {
+	var softwareVersion string
+	softwareInstalled, err := c.GetSoftwareInstalled(ctx)
+	if err != nil {
+		log.Printf("ERROR: couldn't find the softwares installed on array")
+		return 0, err
+	} else {
+		for _, software := range softwareInstalled {
+			if software.IsCluster == true {
+				softwareVersion = software.BuildVersion
+			}
+		}
+	}
+
+	if len(softwareVersion) > 0 {
+		majorVersion, err = strconv.Atoi(softwareVersion[0:1])
+		if err != nil {
+			log.Printf("ERROR: Couldn't convert the software version")
```

Comment:
same here, Change log.Printf to the respective log level type.

Reply:
Updated with log.Errorf statements